### PR TITLE
Create the pid file with integration test

### PR DESCRIPTION
**Pull request by utam0k**
Fix: https://github.com/containers/youki/issues/757

<a href="https://gitpod.io/#https://github.com/containers/youki/pull/762"></a>

### CODE REVIEW

File: crates/libcontainer/src/utils.rs [hunk 1]
```diff
@@ -181,15 +181,10 @@ pub fn secure_join<P: Into<PathBuf>>(rootfs: P, unsafe_path: P) -> Result<PathBu
...
-                part_path = PathBuf::from(part);
-            }
-            None => {
-                break;
-            }
-        }
+        let part_path = match part.next() {
+            None => break,
+            Some(part) => PathBuf::from(part),
+        };
```
Comment (YJDoc2) [inline crates/libcontainer/src/utils.rs:187]: :+1:

File: crates/test_framework/src/test_manager.rs [hunk 2]
```diff
@@ -83,11 +83,10 @@ impl<'a> TestManager<'a> {
...
                 if let Some(tg) = self.test_groups.get(test_group_name) {
-                    let r;
-                    match tests {
-                        None => r = s.spawn(move |_| tg.run_all()),
-                        Some(tests) => r = s.spawn(move |_| tg.run_selected(tests)),
-                    }
+                    let r = match tests {
+                        None => s.spawn(move |_| tg.run_all()),
+                        Some(tests) => s.spawn(move |_| tg.run_selected(tests)),
+                    };
```
Comment (YJDoc2) [inline crates/test_framework/src/test_manager.rs:89]: :+1:

**Review by YJDoc2 — APPROVED** [review]
Looks good to me :+1:
I checked by pulling #758 and applying the patch which allowed it to pass the pidfile test.
Although this makes me wonder why the test was passing originally :thinking:

**Comment by utam0k** [comment]
me too 🤔 
> Although this makes me wonder why the test was passing originally 🤔

**Pull request merged**